crypto/evp/p_lib.c

```diff
@@ -68,7 +68,11 @@ int EVP_PKEY_get_bits(const EVP_PKEY *pkey)
         if (pkey->ameth != NULL && pkey->ameth->pkey_bits != NULL)
             size = pkey->ameth->pkey_bits(pkey);
     }
-    return size < 0 ? 0 : size;
+    if (size <= 0) {
+        ERR_raise(ERR_LIB_EVP, EVP_R_UNKNOWN_BITS);
+        return 0;
+    }
```

Comment:
Errrrrrr....  documentation says:

> **RETURN VALUES**
>   **EVP_PKEY_get_size()**, **EVP_PKEY_get_bits()** and **EVP_PKEY_get_security_bits()** return a positive number, or 0 if this size isn't available.

To me, that indicates that zero is not an error.

Reply:
To me, the doc is kind of ambiguous whether it is an error if the size isn't available.

And reality is, callers of `EVP_PKEY_get_size()` are simply too sloppy.
Just have a look at the output of `grep -A 2 -r EVP_PKEY_get_size providers apps crypto test|fgrep .c`.
Also in `crypto/` and `apps/`, the 0 return value is in most cases not checked at all!
Instead, typically `OPENSSL_malloc()` is called immediately with the value obtained,
this if it is 0, a misleading malloc failure is on the error stack, not indicating an issue with getting the size.

And in those few cases where a 0 return value is checked, no error is put on the queue when bailing out.
This was the motivation for coming up with this PR - 
namely because callers in practice expect that any needed `ERR_raise()` is done by the function itself.